mavisp/modules.py

```diff
@@ -678,6 +678,79 @@ class EnsembleSAS(MavispMultiEnsembleModule, module_class=TaccSAS):
     module_dir = "sas"
     name = "sas"
 
+class MutsOnPhospho(MavispModule):
+    module_dir = "muts_on_phospho"
+    name = "mutsonphospho"
+    expected_files = ['aggregated_filtered_output.csv',
+                      'sasa.rsa']
+
+    sasa_fname = expected_files[1]
+    
+    def _parse_sas(self, fname):
+
+        try:
+            return pd.read_fwf(fname,
+                skiprows=4, skipfooter=4, header=None, widths=[4,4,1,4,9,6,7,6,7,6,7,6,7,6],
+                names = ['entry', 'rest', 'chain', 'resn', 'all_abs', 'sas_all_rel', 'sas_sc_abs',
+                'sas_sc_rel', 'sas_mc_abs', 'sas_mc_rel', 'sas_np_abs', 'sas_np_rel', 'sas_ap_abs',
+                'sas_ap_rel'],
+                usecols = ['resn', 'sas_sc_rel'],
+                index_col = 'resn').fillna(pd.NA)
+
+        except Exception as e:
+            this_error = f"Exception {type(e).__name__} occurred when parsing the sasa.rsa file. Arguments:{e.args}"
+            raise MAVISpMultipleError(warning=warnings,
+                                        critical=[MAVISpCriticalError(this_error)])
+
+    def ingest(self, mutations):
+        
+        warnings = [] 
+
+        aggregated_df = pd.read_csv(os.path.join(self.data_dir, self.module_dir, 'aggregated_filtered_output.csv'))
+        aggregated_df = aggregated_df.drop_duplicates()
+
+        # Load solvent accessibility data using _parse_sas
+        sas_data = self._parse_sas(os.path.join(self.data_dir, self.module_dir, self.sasa_fname))
+
+        gain_of_function = {}
+        loss_of_function = {}
+
+        # Iterate over the aggregated DataFrame to determine gain or loss of function
+        for index, row in aggregated_df.iterrows():
+            for mutation in mutations:
+                mutation_key = mutation
+                restype_resnum_kinase = f"{row['restype']}{row['resnum']}_{row['kinase']}"
```

Comment:
if you do this for every row, it's probably easy just to add a new column before the nested `for`. There are different ways to concatenate text columns in pandas,  but for instance:

```
df = pd.DataFrame({'a':['a','A'], 'b':['b','B'], 'c':['c','C']})
df['a'] + df['b'] + '_' + df['c']
# Out[9]:
# 0    ab_c
# 1    AB_C
```